### PR TITLE
Feat: Add finetuning code & dice loss support for IO

### PR DESCRIPTION
**Pull request by basardemir**

### CODE REVIEW

File: finetuning/README.md [hunk 1]
```diff
@@ -0,0 +1,341 @@
...
+```bash
+python finetune.py --config configs/my_config.yaml
+```
+
+### Step 4: Monitor Training
+
+Training progress is logged to TensorBoard:
+
+```bash
+tensorboard --logdir=results/my_finetuning_experiment
```
Comment (lintian-a) [inline finetuning/README.md]: Are the log files saved in the specified output folder, such as " tensorboard --logdir=my_config[training][output_folder]/my_finetuning_experiment"?

File: finetuning/README.md [hunk 2]
```diff
@@ -0,0 +1,341 @@
...
+- [Configuration Guide](#-configuration-guide)
+- [Dataset Types](#-dataset-types)
+- [Advanced Features](#-advanced-features)
+
+## 🚀 Quick Start
+
+```bash
+# Navigate to finetuning directory
+cd uniGradICON/
+pip install .
```
Comment (lintian-a) [inline finetuning/README.md]: Do we need to consider users who install using "pip install unigradicon"?

File: finetuning/config_loader.py [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+            "(unpaired_with_seg, paired_with_seg)."
+        )
+    
+    if has_seg:
+        return 'segmentation'
+    else:
+        return 'standard'
+
+
+def create_dataset_from_config(dataset_config: Dict[str, Any], input_shape: Tuple[int, ...]) -> dataset.Dataset:
```
Comment (lintian-a) [inline finetuning/config_loader.py]: Do we need to validate that the image files in the dataset JSON file exist?

File: finetuning/dataset.py [hunk 4]
```diff
@@ -0,0 +1,399 @@
...
+
+    if hasattr(itk, "AnatomicalOrientation"):
+        desired_coordinate_orientation = itk.AnatomicalOrientation(desired_coordinate_orientation)
+
+    return itk.orient_image_filter(
+        moving, 
+        desired_coordinate_orientation=desired_coordinate_orientation,
+        use_image_direction=True)
+
+class Dataset:
```
Comment (lintian-a) [inline finetuning/dataset.py]: Is this Dataset class inherited from torch.utils.data.Dataset?

File: src/unigradicon/finetuning/multi_dataset_loader.py [hunk 5]
```diff
@@ -0,0 +1,118 @@
...
+        )
+    
+    combined_dataset = ConcatDataset(datasets)
+    total_samples = len(combined_dataset)
+    print(f"\nTotal combined samples: {total_samples}")
+    
+    samples_per_epoch = config['training'].get('samples_per_epoch', total_samples)
+    
+    total_weight = sum(weights)
+    normalized_weights = [w / total_weight for w in weights]
```
Comment (lintian-a) [inline src/unigradicon/finetuning/multi_dataset_loader.py]: Do we want the weight to balance the sizes of the datasets, namely, by sampling more data from the smaller dataset? The current implementation seems to assign a lower weight to datasets of smaller size.
